test/controllers/public_controller_test.rb

```diff
@@ -25,4 +41,22 @@ class PublicControllerTest < ActionDispatch::IntegrationTest
     get terms_of_use_url
     assert_response :success
   end
+
+  test "should get contact_us" do
+    get contact_us_url
+    assert_response :success
+  end
+
+  test "should post contact_us and save information about customer request" do
+    assert_difference('ContactRequest.count') do 
+      post contact_us_url, params: @contact_us_params
+    end
+    assert_response :success
+  end
+
+  test 'should send email to admins' do
+    assert_difference('ActionMailer::Base.deliveries.count') do
+      post contact_us_url, params: @contact_us_params
+    end
+  end
```

Comment:
Please add tests for failed model validations

Reply:
I meant the controller tests for failed model validation - invalid contact params, e.g. missing email, but yeah, my comment was not clear enough

Reply:
Added.